### PR TITLE
Use the search order settings for the search editor

### LINKED ISSUE #103625: Search in the search editor windows does not respect the order field
- VSCode Version: 1.47.3
- OS Version: macOS

Steps to Reproduce:

1. Settings > "Search: Sort Order" > select "countDescending"
2. Ctrl+P "> Search editor" (open the search editor)
3. Search something, results appear sorted by path

Does this issue occur when all extensions are disabled?: Yes

This is due to the search editor not loading the configuration.

I already have a fix for that and will send a PR.

### PR DESCRIPTION
**Pull request by damienmg**
Before this change searches in the search editor would always
be sorted with the default sort, now it respect the configuration
in the search settings.

This PR fixes #103625.

### CODE REVIEW

**Comment by damienmg** [comment]
I cannot make sense of the test failures, are those tests flaky?

File: src/vs/workbench/contrib/searchEditor/browser/searchEditorSerialization.ts [hunk 1]
```diff
@@ -208,7 +208,7 @@ export const extractSearchQueryFromLines = (lines: string[]): SearchConfiguratio
 };
 
 export const serializeSearchResultForEditor =
-	(searchResult: SearchResult, rawIncludePattern: string, rawExcludePattern: string, contextLines: number, labelFormatter: (x: URI) => string): { matchRanges: Range[], text: string, config: Partial<SearchConfiguration> } => {
+	(searchResult: SearchResult, rawIncludePattern: string, rawExcludePattern: string, contextLines: number, labelFormatter: (x: URI) => string, sortOrder: SearchSortOrder = SearchSortOrder.Default): { matchRanges: Range[], text: string, config: Partial<SearchConfiguration> } => {
```
Comment (JacksonKearl) [inline src/vs/workbench/contrib/searchEditor/browser/searchEditorSerialization.ts:211]: I'd remove the default parameter on `sortOrder`, instead it should be either required, or optional so the fallback occurs at the existing `searchMatchComparer`, that way if it's ever changed in the future we just make one change.

**Review by JacksonKearl — CHANGES_REQUESTED** [review]
Looks good overall, just a small nit.

File: src/vs/workbench/contrib/searchEditor/browser/searchEditorSerialization.ts (hunk 1 above)
Comment (damienmg) [reply src/vs/workbench/contrib/searchEditor/browser/searchEditorSerialization.ts]: I made it required since we provide it in all the place.

**Comment by JacksonKearl** [comment]
Looks good, thanks for adding this!

**Comment by JacksonKearl** [comment]
And yes, the tests are flaky... 🙈🙉🙊

**Pull request merged**